src/platform/plugins/shared/data/public/search/session/session_service.tsx

```diff
@@ -704,6 +705,25 @@ export class SessionService {
     };
   }
 
+  public getBackgroundSearchIndicator() {
+    // const bgSearch = useObservable(this.state$);
+    const bgSearchEnabled = this.featureFlags?.getBooleanValue(
+      BACKGROUND_SEARCH_FEATURE_FLAG_KEY,
+      false
+    );
+
+    if (!bgSearchEnabled) {
+      return null;
+    }
+    // Note that there are more bgSearch states we need to check for
+    return (
+      <EuiCallOut size="s">
+        You are viewing cached data from a specific time range from a background search. Changing
+        the time range or query will re-run the session.
+      </EuiCallOut>
```

Comment:
@AlexGPlay Part 2, this just works because I converted the file to React / TSX, POC style. but the question would be how to make it right, so it's available in Discover / Dashboards.